### PR TITLE
fix: object store metrics collection for aggregate

### PR DESCRIPTION
**Pull request by rahulguptajss**

### CODE REVIEW

File: cmd/collectors/rest/plugins/aggregate/aggregate.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		}
+	}
+
+	instanceKeys.NewChildS("", "bin_num")
+	instanceKeys.NewChildS("", "tier")
+	mat.SetExportOptions(exportOptions)
+
+	for _, k := range metrics {
+		if err := matrix.CreateMetric(k, mat); err != nil {
+			a.Logger.Warn().Err(err).Str("key", k).Msg("error while creating metric")
```
Comment (cgrinds) [inline cmd/collectors/rest/plugins/aggregate/aggregate.go:75]: no need to double log

File: cmd/collectors/rest/plugins/aggregate/aggregate.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	return mat, nil
+}
+
+func (a *Aggregate) Run(dataMap map[string]*matrix.Matrix) ([]*matrix.Matrix, *util.Metadata, error) {
+	data := dataMap[a.Object]
+	a.client.Metadata.Reset()
+
+	aggrSpaceMat, err := a.initMatrix(aggrObjectStoreMatrix, data)
+	if err != nil {
+		a.Logger.Warn().Err(err).Msg("error while initializing matrix")
```
Comment (cgrinds) [inline cmd/collectors/rest/plugins/aggregate/aggregate.go:89]: no need to double log

File: cmd/collectors/rest/plugins/aggregate/aggregate.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		uuid, has := uuidLookup[aggrName]
+		if !has {
+			continue
+		}
+
+		binNum := record.Get("bin_num").String()
+		tierName := record.Get("tier_name").String()
+		logicalUsed := record.Get("object_store_logical_used").String()
+		physicalUsed := record.Get("object_store_physical_used").String()
+		instanceKey := fmt.Sprintf("%s_%s_%s", aggrName, tierName, binNum)
```
Comment (cgrinds) [inline cmd/collectors/rest/plugins/aggregate/aggregate.go:134]: minor: it's more efficient to use `aggrName + "_" + tierName + "_" + binNum`

File: cmd/collectors/rest/plugins/aggregate/aggregate.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+		}
+	}
+}
+
+func (a *Aggregate) getObjectStoreData() ([]gjson.Result, error) {
+	fields := []string{"aggregate_name", "bin_num", "tier_name", "object_store_logical_used", "object_store_physical_used"}
+	href := rest.NewHrefBuilder().
+		APIPath(apiQuery).
+		Fields(fields).
+		Filter([]string{"tier_name=!\" \"|\"\""}).
```
Comment (cgrinds) [inline cmd/collectors/rest/plugins/aggregate/aggregate.go:165]: a little bit easier to read using raw literals
`tier_name=!" "|""`

**Review by cgrinds — COMMENTED** [review]
Looks like the new aggregate counters are only available via REST. We should add a note at the top of the dashboard to clarify that to avoid confusion on why these new panels have no data when using the Zapi collector.

**Review by Hardikl — APPROVED** [review]

**Review by cgrinds — APPROVED** [review]

**Pull request merged**